### PR TITLE
[JENKINS-60866][JENKINS-71797] Remove timeline widget

### PR DESCRIPTION
**Pull request by daniel-beck**
See [JENKINS-60866](https://issues.jenkins.io/browse/JENKINS-60866).

This removes the timeline widget without replacement. I've never found it particularly useful, and it is a problem for CSP. We probably want something different for these views, but for now, remove without replacement.

<details>
<summary>Screenshots</summary>
Screenshot 2023-08-28 at 19 31 04

Screenshot 2023-08-28 at 19 31 14

Screenshot 2023-08-28 at 19 31 46

</details>

### Proposed changelog entries

* Remove build timeline widget from the build history pages of views, jobs, and agents.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")` if applicable.
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/6869"></a>

### CODE REVIEW

**Review by timja — COMMENTED** [review]
Makes sense to me, is there anything remaining to take this out of draft?

See also discussion on the forum:
https://community.jenkins.io/t/does-anybody-find-the-build-history-timeline-useful/8491/5

**Comment by daniel-beck** [comment]
Noting https://github.com/jenkinsci/jqs-monitoring-plugin/blob/f50a82a0bf323c647c68aadd9eda9590a9468c2d/src/main/java/org/jenkinsci/plugins/jqsmonitoring/slaves/SlavesHolder.java#L109 / https://github.com/jenkinsci/jqs-monitoring-plugin/blob/f50a82a0bf323c647c68aadd9eda9590a9468c2d/src/main/java/org/jenkinsci/plugins/jqsmonitoring/slaves/SlavesHolder.java#L124 as API user who will not be affected by this change (other than a `DoNotUse` error once the core dependency is sufficiently high).

**Comment by daniel-beck** [comment]
Might even fix (or alleviate somewhat) https://issues.jenkins.io/browse/JENKINS-72138

**Comment by timja** [comment]
Could you run this through ATH and bom please to be safe? Happy to approve after that.

**Review by timja — APPROVED** [review]
LGTM

**Review by rsahoo7495 — APPROVED** [review]
LGTM

**Comment by MarkEWaite** [comment]
+1 from me to remove the widget.  That should resolve the JavaScript warning that I see in the browser console when viewing the timeline.  The message in the console is:

```
GET https://www.example.com/jenkins/view/Controllers/__history__.html?0 404 (Not Found)
SimileAjax.History.initialize @ simile-ajax-bundle.js:921
SimileAjax.History.addListener @ simile-ajax-bundle.js:926
SimileAjax.WindowManager.initialize @ simile-ajax-bundle.js:1360
Timeline._Impl @ timeline-bundle.js:2619
Timeline.create @ timeline-bundle.js:2556
doLoad @ build-timeline-widget.js:80
load (async)
(anonymous) @ build-timeline-widget.js:109
```

**Comment by timja** [comment]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!
